### PR TITLE
Feature 312 support for theming envs

### LINKED ISSUE #312: Support for theming envs
As a developer i would like to add support for theming envs, to allow easier and more flexible theming.

### PR DESCRIPTION
**Pull request by Valmetti**
Resolves #312 
Requires additional envs for testing.

### CODE REVIEW

File: src/components/layout-components/header.tsx [hunk 1]
```diff
@@ -106,22 +116,25 @@ const Header: React.FC = ({
...
     >
       <MenuItem style={{ backgroundColor: "#fff" }}>{ renderLogoutButton() }</MenuItem>
     </HeaderSelect>
   );
 
   return (
-    <Root position="static">
+    <Root
+      position="static"
+      sx={{ backgroundColor: process.env.REACT_APP_THEME_USE_COLORED_HEADER === "true" ? process.env.REACT_APP_THEME_PALETTE_SECONDARY : "" }}
```
Comment (EVP1989) [inline src/components/layout-components/header.tsx:128]: Should this logic also be in theme file. Starts to look bit messy if it's used in the component.

Comment (EVP1989) [reply src/components/layout-components/header.tsx:128]: Same question for the rest of the file

File: src/styled/layouts/basic-layout.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const Root = styled(Box, {
   display: "grid",
   gridTemplateRows: "auto auto auto 1fr",
   backgroundColor: theme.palette.background.paper,
-  backgroundImage: `url(${theme.backgroundImage.backgroundImagePath})`,
+  backgroundImage: `url(${process.env.REACT_APP_THEME_BACKGROUNDIMAGE_PATH})`,
```
Comment (EVP1989) [inline src/styled/layouts/basic-layout.ts:17]: Move all the envs to config.ts and make functions to use them. This applies to whole pr, replace all "process.env..." to be a getter method.

File: src/theme/index.ts [hunk 3]
```diff
@@ -6,55 +6,80 @@ import { createTheme, darkScrollbar } from "@mui/material";
 const { breakpoints, palette } = createTheme();
 
 /**
- * Custom theme for Material UI
+ * Converts color hex value to RGB value
  */
-export default createTheme({
+function hexToRgb(color: string): { r: number; g: number; b: number } {
```
Comment (EVP1989) [inline src/theme/index.ts:11]: Move the functions to a theme-utils class

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: public/index.html [hunk 4]
```diff
@@ -32,7 +32,7 @@
       rel="stylesheet"
       href="https://fonts.googleapis.com/icon?family=Material+Icons"
     />
-    <title>Eloisa Management Portal</title>
+    <title>Metaform Portal</title>
```
[the remark this answers is not in the document]
Comment (Valmetti) [reply public/index.html:35]: The theme title replaces this one, this is a backup if there is no title in theme.

File: src/theme/index.ts [hunk 5]
```diff
@@ -10,51 +12,35 @@ const { breakpoints, palette } = createTheme();
...
-
   sectionTitle: {
-    fontFamily: "Arial, sans-serif",
+    fontFamily: Config.get().theme.fontFamily,
     fontWeight: 200,
     fontSize: 26
   },
 
+  header: {
+    backgGround: ThemeUtils.getHeaderBackgroundColor(),
```
Comment (EVP1989) [inline src/theme/index.ts:22]:
```suggestion
    background: ThemeUtils.getHeaderBackgroundColor(),
```

File: src/utils/theme-utils.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    return {
+      r: r,
+      g: g,
+      b: b
+    };
+  };
+
+  /**
+   * Gets color brightness
+   * @param r red value
```
Comment (EVP1989) [inline src/utils/theme-utils.ts:24]:
```suggestion
   * Gets color brightness
   *
   * @param r red value
```

File: src/utils/theme-utils.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+   * @param g green value
+   * @param b blue value
+   */
+  export const calculateBrightness = ({ r, g, b }: { r: number; g: number; b: number }) => {
+    return (r * 299 + g * 587 + b * 114) / 1000;
+  };
+
+  /**
+   * Gets the contrast text color based on the background color
+   * @param backgroundColor background color
```
Comment (EVP1989) [inline src/utils/theme-utils.ts:34]:
```suggestion
   * Gets the contrast text color based on the background color
   *
   * @param backgroundColor background color
```

File: src/utils/theme-utils.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  export const getContrastTextColor = (backgroundColor: string) => {
+    const rgb = hexToRgb(backgroundColor);
+    const brightness = calculateBrightness(rgb);
+    
+    return brightness > 128 ? "#000000" : "#FFFFFF";
+  };
+
+  /**
+   * Lightens a color by a percentage
+   * @param color color hex value
```
Comment (EVP1989) [inline src/utils/theme-utils.ts:45]:
```suggestion
   * Lightens a color by a percentage
   *
   * @param color color hex value
```

File: src/utils/theme-utils.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+      g: Math.round((255 - rgb.g) * percent) + rgb.g,
+      b: Math.round((255 - rgb.b) * percent) + rgb.b
+    };
+
+    return `rgb(${newRgb.r}, ${newRgb.g}, ${newRgb.b})`;
+  };
+
+  /**
+   * Returns the background color of the header
+   * @returns The background color of the header
```
Comment (EVP1989) [inline src/utils/theme-utils.ts:61]:
```suggestion
   * Returns the background color of the header
   *
   * @returns The background color of the header
```

**Review by EVP1989 — APPROVED** [review]
Approved with suggestions

**Pull request merged**